ProcessList.c

```diff
@@ -194,308 +191,141 @@ void ProcessList_remove(ProcessList* this, const Process* p) {
-      proc->tree_index = idx;
-      proc->tree_depth = deep;
-
-      if (indent == -1) {
-         proc->indent = 0;
-      } else if (i == size - 1) {
-         proc->indent = -currentIndent;
+   // Find the end to know the last line for indent handling purposes
+   int lastShown = r;
+   while ((r < vsize)) {
+      Process* process = (Process*)Vector_get(this->processes, r);
+      if (Process_isChildOf(process, pid)) {
+         if (process->show)
+            lastShown = r;
+         r++;
       } else {
-         proc->indent = currentIndent;
+         break;
       }
-
-      Hashtable_put(this->draftingTreeSet, proc->tree_index, proc);
-
-      // It's not strictly necessary to do this, but doing so anyways
-      // allows for checking the correctness of the inner workings.
-      Hashtable_remove(this->displayTreeSet, newLeftBound);
    }
 
-   Vector_delete(layer);
-}
-
-static void ProcessList_updateTreeSet(ProcessList* this) {
-   unsigned int index = 0;
-   unsigned int tree_index = 1;
-
-   const int vsize = Vector_size(this->processes);
-
-   assert(Hashtable_count(this->draftingTreeSet) == 0);
-   assert((int)Hashtable_count(this->displayTreeSet) == vsize);
-
-   ProcessList_updateTreeSetLayer(this, 0, vsize, 0, 0, vsize * 2 + 1, &index, &tree_index, -1);
-
-   Hashtable* tmp = this->draftingTreeSet;
-   this->draftingTreeSet = this->displayTreeSet;
-   this->displayTreeSet = tmp;
-
-   assert(Hashtable_count(this->draftingTreeSet) == 0);
-   assert((int)Hashtable_count(this->displayTreeSet) == vsize);
-}
-
-static void ProcessList_buildTreeBranch(ProcessList* this, pid_t pid, int level, int indent, int direction, bool show, int* node_counter, int* node_index) {
-   // On OpenBSD the kernel thread 'swapper' has pid 0.
-   // Do not treat it as root of any tree.
-   if (pid == 0)
-      return;
-
-   Vector* children = Vector_new(Class(Process), false, DEFAULT_SIZE);
-
-   for (int i = Vector_size(this->processes) - 1; i >= 0; i--) {
+   for (int i = l; i < r; i++) {
       Process* process = (Process*)Vector_get(this->processes, i);
-      if (process->show && Process_isChildOf(process, pid)) {
-         process = (Process*)Vector_take(this->processes, i);
-         Vector_add(children, process);
-      }
-   }
-
-   int size = Vector_size(children);
-   for (int i = 0; i < size; i++) {
-      int index = (*node_index)++;
-      Process* process = (Process*)Vector_get(children, i);
-
-      int lft = (*node_counter)++;
 
       if (!show) {
          process->show = false;
       }
 
-      int s = Vector_size(this->processes2);
-      if (direction == 1) {
-         Vector_add(this->processes2, process);
-      } else {
-         Vector_insert(this->processes2, 0, process);
-      }
+      int s = Vector_size(this->displayList);
+      Vector_add(this->displayList, process);
 
-      assert(Vector_size(this->processes2) == s + 1); (void)s;
+      assert(Vector_size(this->displayList) == s + 1); (void)s;
```

Comment:
I would argue this is not necessary, by now `Vector_add()` has proven to be working.

Reply:
This check is a consistency check for the tree sorting code.

Reply:
Done. The consistency check value of this specific assert was really low.